tests/cpp/precommit/test_model_config.cpp

```diff
@@ -0,0 +1,117 @@
+#include <stddef.h>
+
+#include <cstdint>
+#include <exception>
+#include <iomanip>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <fstream>
+
+#include <nlohmann/json.hpp>
+
+#include <opencv2/core.hpp>
+
+#include <gtest/gtest.h>
+
+#include <models/classification_model.h>
+#include <models/detection_model.h>
+#include <models/input_data.h>
+#include <models/results.h>
+
+using json = nlohmann::json;
+
+std::string DATA_DIR = "../data";
+std::string MODEL_PATH_TEMPLATE = "public/%s/FP16/%s.xml";
+std::string IMAGE_PATH = "coco128/images/train2017/000000000074.jpg";
+
+std::vector<std::string> model_list = {"efficientnet-b0-pytorch"};
+
+struct ModelData {
+    std::string name;
+    ModelData(const std::string& name)
+        : name(name) {}
+};
+
+class ModelParameterizedTest : public testing::TestWithParam<ModelData> {
+};
+
+template<typename... Args>
+std::string string_format(const std::string &fmt, Args... args)
+{
+    size_t size = snprintf(nullptr, 0, fmt.c_str(), args...);
+    std::string buf;
+    buf.reserve(size + 1);
+    buf.resize(size);
+    snprintf(&buf[0], size + 1, fmt.c_str(), args...);
+    return buf;
+}
+
+// TODO: Add tests for create_model
+ 
+TEST_P(ModelParameterizedTest, EfficientnNetTest)
+{
+    auto model_path = string_format(MODEL_PATH_TEMPLATE, GetParam().name.c_str(), GetParam().name.c_str());
+    auto model = ClassificationModel::create_model(DATA_DIR + "/" + model_path);
+    
+    auto ov_model = model->getModel();
+
+    ov::serialize(ov_model, "tmp_model.xml");
```

Comment:
We will need to revise how tests are organized. I see the same problem with introducing tests for otx model.
for now maybe you should move this to `test_sanity.cpp`. You introduce lots of code copied from `test_sanity.cpp` just to test `model->getModel();` and `ov::serialize()`.

Reply:
That is a draft of the test. The plan is to fill this file with many tests.